storage_controller/src/node.rs

```diff
@@ -87,6 +91,11 @@ impl Node {
             && self.availability_zone_id == register_req.availability_zone_id
     }
 
+    // Do we need to update an existing record in DB on this registration request?
+    pub(crate) fn need_update(&self, register_req: &NodeRegisterRequest) -> bool {
+        self.listen_https_port != register_req.listen_https_port
```

Comment:
It's worth a comment here and in `registration_match()` why we check this here and not in `registration_match`. If the plan is to move this into `registration_match` once the migration is complete (I think we should) then it's also worth spelling out that plan.